syntax = "proto3";

package vision_service;

// 检测结果
message DetectionResult {
  string id = 1;
  string skill_name = 2;
  repeated Detection detections = 3;
  string image_url = 4;
  string video_url = 5;
  string alert_level = 6;
  int64 timestamp = 7;
}

// 单个检测项
message Detection {
  string class_name = 1;
  float confidence = 2;
  BoundingBox bbox = 3;
  map<string, string> attributes = 4;
}

// 边界框
message BoundingBox {
  float x = 1;
  float y = 2;
  float width = 3;
  float height = 4;
}

// 图片检测请求
message ImageDetectionRequest {
  bytes image_data = 1;
  string skill_name = 2;
  string alert_level = 3;
  repeated string roi = 4;
}

// 视频流检测请求
message VideoStreamRequest {
  string stream_url = 1;
  string skill_name = 2;
  string alert_level = 3;
  int32 frame_interval = 4;
  repeated string roi = 5;
  string schedule = 6;
}

// 检测响应
message DetectionResponse {
  string request_id = 1;
  string status = 2;
  string message = 3;
}

// 视觉AI服务
service VisionService {
  // 图片检测
  rpc DetectImage(ImageDetectionRequest) returns (DetectionResponse) {}
  
  // 视频流检测
  rpc DetectVideoStream(VideoStreamRequest) returns (DetectionResponse) {}
} 